qiita_pet/templates/study_ajax/prep_summary.html

```diff
@@ -1,115 +1,407 @@
-    <button class="btn btn-danger btn-sm" onclick="prep_action('delete')">Delete</button>
-    {% end %}
-    </p>
-</div>
+  /*
+   * Function to draw a vis.Network
+   *
+   * @param nodes list of nodes
+   * @param edges list of edges
+   * @param target div name to draw the netword
+   * @param on_select_f callback function for the 'select' event in the network nodes
+   *
+   * This function creates a new instance of vis.Network and draws the graph
+   * in the target div
+   *
+   */
+  function draw_graph(nodes, edges, target, on_select_f) {
+    var container = document.getElementById(target);
+    container.innerHTML = "";
+    var data = {
+      nodes: nodes,
+      edges: edges
+    };
+    var options = {
+      nodes: {
+        color: '#ffffff',
+        font: '16px arial black'
+      },
+      layout : {
+        hierarchical: {
+          direction: "LR",
+          sortMethod: "directed",
+          levelSeparation: 260
+        }
+      },
+      interaction: {
+        dragNodes: false,
+        dragView: false,
+        zoomView: false
+      }
+    };
+    network = new vis.Network(container, data, options);
 
-<div>
-<h3>Update Investigation Type</h3>
-    <p id="ena-info"> Selected investigation type: 
-    <select id="ena-ontology" value="ena-ontology">
-    <option value=""></option>
-    {% for o in ontology['ENA'] %}
-      <option value="{{o}}">{{o}}</option>
-    {% end %}
-      <option value="Other">Other</option>
-    </select></p>
-
-    <p id="user-ena-info"> User defined investigation type: 
-    <select id="user-ontology" value="user-ontology">
-    <option value=""></option>
-    {% for o in ontology['User'] %}
-      <option value="{{o}}">{{o}}</option>
-    {% end %}
-    <option value="New Type">New Type</option>
-    </select></p>
-
-    <p id="new-ena-info">New user defined term: 
-    <input type="textbox" id="new-ontology" name="new-ontology"></p>
-
-    <button class="btn btn-info btn-sm" onclick="prep_action('ontology')">Update</button>
-</div>
+    network.on('select', on_select_f);
+  }
 
-<div>
-    <h3>Prep information summary</h3>
-    <p>There are <strong>{{num_samples}}</strong> samples in this study.</p>
-</div>
+  /*
+   * Function to update the prep information contents
+   *
+   * This function executes an AJAX query to modify the contents of the
+   * prep information and updates the interface accordingly
+   *
+   */
+  function update_prep_information() {
+    var fp = $("#file-selector").val();
+    $.ajax({
+      url: '/prep_template/',
+      type: 'PATCH',
+      data: {'op': 'replace', 'path': '/{{prep_id}}/data', 'value': fp},
+      success: function(data) {
+        if(data.status == 'error') {
+          bootstrapAlert(data.message, "danger");
+        }
+        else {
+          if(data.status == 'warning') {
+            bootstrapAlert(data.message.replace("\n", "<br/>"), "warning");
+          }
+          $("#file-selector option[value='" + fp +"']").remove();
```

Comment:
Interesting, well, the problem for the most part is that if the string
in a query command in jQuery contains special characters, the query is
likely to fail, but there's no chance we'll see special characters here
then.

On (Mar-14-16|22:14), Jose Navas wrote:
> > +   */
> > +  function update_prep_information() {
> > +    var fp = $("#file-selector").val();
> > +    $.ajax({
> > +      url: '/prep_template/',
> > +      type: 'PATCH',
> > +      data: {'op': 'replace', 'path': '/{{prep_id}}/data', 'value': fp},
> > +      success: function(data) {
> > +        if(data.status == 'error') {
> > +          bootstrapAlert(data.message, "danger");
> > +        }
> > +        else {
> > +          if(data.status == 'warning') {
> > +            bootstrapAlert(data.message.replace("\n", "<br/>"), "warning");
> > +          }
> > +          $("#file-selector option[value='" + fp +"']").remove();
> 
> Probably is a bit hard to read, but the `fp` is actually contained in a javascript string, so I think it shouldn't affect and I didn't had any problem while I was testing this interface:
> 
> ```JavaScript
> $("#file-selector option[value='some_file_with.reserved.chars']").remove();
> ```
> 
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1683/files#r56115939